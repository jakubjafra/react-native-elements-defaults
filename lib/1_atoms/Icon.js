const _ = require('lodash');
const utils = require('../utils');

// TODO: Make this function immutable.
function override(defaultStyles) {
    const __oldModule__ = require('react-native-elements/src/icons/Icon');
    const __oldModuleDefault__ = __oldModule__.default;

    const Button = function (props) {
        const newProps = {};

        newProps.style = utils.overrideBottom(props.style, defaultStyles.style);
        newProps.iconStyle = utils.overrideBottom(props.iconStyle, defaultStyles.iconStyle);
        newProps.containerStyle = utils.overrideBottom(props.containerStyle, defaultStyles.containerStyle);

        return __oldModuleDefault__.call(undefined, _.defaultsDeep(newProps, props));
    };

    Button.propTypes = __oldModuleDefault__.propTypes;
    __oldModule__.default = Button;
}

module.exports = override;
