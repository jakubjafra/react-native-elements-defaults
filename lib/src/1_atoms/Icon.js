const _defaults = require('lodash.defaults');
const utils = require('../utils');

function compose(defaultStyles) {
    return function(__oldModuleDefault__) {
        const Icon = function(props) {
            const newProps = {};

            newProps.style = utils.overrideBottom(props.style, defaultStyles.style);
            newProps.iconStyle = utils.overrideBottom(props.iconStyle, defaultStyles.iconStyle);
            newProps.containerStyle = utils.overrideBottom(props.containerStyle, defaultStyles.containerStyle);

            return __oldModuleDefault__.call(undefined, _defaults(newProps, props));
        };

        Icon.propTypes = __oldModuleDefault__.propTypes;
        Icon.defaultProps = __oldModuleDefault__.defaultProps;

        return Icon;
    }
}

module.exports = {
    compose,
    override: new utils.ModuleOverride('atoms/icon', require('react-native-elements/src/icons/Icon'), compose),
};
