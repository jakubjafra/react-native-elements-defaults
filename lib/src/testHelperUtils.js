import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name,
                      __oldElement__,
                      __element__,
                      basicTheme,
                      createElement = element => React.createElement(element, null, null)) {
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
