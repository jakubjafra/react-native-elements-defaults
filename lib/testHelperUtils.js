import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name, __oldElement__, __element__, basicTheme){
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(<OldElement />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(<Element />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(<OldElement />).toJSON();
            const tree = renderer.create(<Element />).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(<OldElement />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(<Element />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(<OldElement />).toJSON();
            const tree = renderer.create(<Element />).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
