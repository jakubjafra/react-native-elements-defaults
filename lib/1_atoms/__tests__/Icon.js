import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import * as utils from '../../testHelperUtils';

describe('Icon', () => {
    utils.defaultTests(
        'Icon',
        require('react-native-elements/src/icons/Icon'),
        require('../Icon'),
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );
});
