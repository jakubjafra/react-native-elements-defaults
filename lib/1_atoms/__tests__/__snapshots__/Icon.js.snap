// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon with basic theme applied Icon renders correctly 1`] = `
<View
  onPress={undefined}
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  underlayColor="white"
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon with basic theme applied OldIcon renders correctly 1`] = `
<View
  onPress={undefined}
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon with no theme applied Icon renders correctly 1`] = `
<View
  onPress={undefined}
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon with no theme applied OldIcon and Icon returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`Icon with no theme applied OldIcon renders correctly 1`] = `
<View
  onPress={undefined}
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
